editor/inserter/menu.js

```diff
@@ -283,7 +306,7 @@ export class InserterMenu extends Component {
 				role="menuitem"
 				key={ block.name }
 				className="editor-inserter__block"
-				onClick={ this.selectBlock( block.name ) }
+				onClick={ this.selectBlock( block.name, block.attributes ) }
```

Comment:
I find it a bit odd at a glance that in arguments to "selecting" a block we pass attributes. What does attributes have to do with a selection? Seems like this is a convenience for something we could be looking up later in the logic to map that selection into a block creation?

Reply:
Yeah, that's a good point. I'll play with making `selectBlock` pass up an _enum-ish_ object (e.g. `{ name: 'core/paragraph' }` or `{ name: 'core/reusable-block', ref: '6c08c662-3825-4f9b-9512-ff8a2d209a6b' }`) instead of a string.

Reply:
I think the right thing to do here is to refactor `InserterMenu` so that it no longer calls `getCategories`, `getBlockTypes`, `getBlocks`, etc. Instead, `InserterMenu` would be a "dumb" component that renders a list of `items` that `Inserter` passes it and calls `item.onSelect()` when one of these items is selected.

Some code to illustrate what I mean:

```js
const staticItems = getBlockTypes()
	.filter( ( blockType ) => ! blockType.isPrivate )
	.map( ( blockType ) => ( {
		category: blockType.category,
		icon: blockType.icon,
		title: blockType.title,
		isDisabled: blockType.useOnce && find( blocks, ( { name } ) => blockType.name === name ),
		isRecent: find( recentlyUsedBlocks, ( { name } ) => blockType.name === name ),
		onSelect() {
			onInsertBlock( blockType.name, {}, insertionPoint );
			onClose();
		},
	} ) );

const reusableItems = /* omitted for brevity */;

return (
	<InserterMenu
		categories={ getCategories() }
		items={ [ ...staticItems, ...reusableItems ] } />
);
```

My fear though that this is quite a lot to change in this already large PR—I'd prefer to ticket this and come back to it later.